src/common/type/blob.h

```diff
@@ -0,0 +1,58 @@
+/***********************************************************************************************************************************
+Blob Handler
+
+Efficiently store large numbers of small allocations by packing them onto larger blocks. Note that the allocations will not be
+aligned unless all of them are aligned.
+
+A usage example is to store zero-terminated strings, which do not need to be aligned, and where alignment might waste significant
+space.
+
+Note that this code is fairly primitive, and this should probably be a capability of memory contexts. However, for now it reduces
+memory requirements for large numbers of zero-terminated strings.
+***********************************************************************************************************************************/
+#ifndef COMMON_TYPE_BLOB_H
+#define COMMON_TYPE_BLOB_H
+
+/***********************************************************************************************************************************
+Size of blocks allocated for blob data
+***********************************************************************************************************************************/
+#ifndef BLOB_BLOCK_SIZE
+    #define BLOB_BLOCK_SIZE                                         (64 * 1024)
+#endif
+
+/***********************************************************************************************************************************
+Object type
+***********************************************************************************************************************************/
+typedef struct Blob Blob;
+
+#include "common/type/object.h"
+
+/***********************************************************************************************************************************
+Constructor
+***********************************************************************************************************************************/
+Blob *blbNew(void);
+
+/***********************************************************************************************************************************
+Functions
+***********************************************************************************************************************************/
+// Add data to the blob
```

Comment:
Since this is used mostly for efficient storage of strings, maybe we should have a "blbAddStr(*blob, *char) convenience function.

Reply:
I thought about that, but would involve an extra call to strlen(). In this case the caller already knows the string size. Better might be blbAddStr(Blob *, String *) but that would be confusing because we would not being storing the entire string object, just the buffer holding the zero-terminated string.

Reply:
Someday, somebody (maybe me), is going to forget to add 1 to the strlen.  But you are right about efficiency and confusion.

Reply:
I actually did forget to add 1 on my first pass and it blew up in a pretty spectacular fashion. Not only did the tests fail completely, valgrind complained that I had read into unitialized memory (since it never stopped reading from the first string). So, it's a concern but I think we have enough tools in place to catch that type of error.